wallet/src/main/scala/org/bitcoins/wallet/internal/UtxoHandling.scala

```diff
@@ -39,6 +39,48 @@ private[wallet] trait UtxoHandling extends WalletLogger {
     spendingInfoDAO.findAllUnspentForAccount(hdAccount)
   }
 
+  protected def updateUtxoConfirmedState(
+      txo: SpendingInfoDb,
+      blockHash: DoubleSha256DigestBE): Future[SpendingInfoDb] = {
+    for {
+      confsOpt <- chainQueryApi.getNumberOfConfirmations(blockHash)
+      stateChange <- {
+        confsOpt match {
+          case None => Future.successful(txo)
+          case Some(confs) =>
+            txo.state match {
+              case TxoState.PendingConfirmationsReceived |
+                  TxoState.DoesNotExist =>
+                if (confs >= walletConfig.requiredConfirmations) {
+                  spendingInfoDAO.update(
+                    txo.copyWithState(TxoState.ConfirmedReceived))
+                } else {
+                  Future.successful(
+                    txo.copyWithState(TxoState.PendingConfirmationsReceived))
+                }
+              case TxoState.PendingConfirmationsSpent =>
+                if (confs >= walletConfig.requiredConfirmations) {
+                  spendingInfoDAO.update(
+                    txo.copyWithState(TxoState.ConfirmedSpent))
+                } else {
+                  Future.successful(txo)
+                }
+              case TxoState.Reserved =>
+                // We should keep the utxo as reserved so it is not used in
```

Comment:
but doesn't this really mean we have seen a utxo spent on the network that we thought we had reserved? Isn't this bad? Worse case couldn't this be a hack, best case somehow how we are out of sync?

Reply:
Yeah, you're right, this should be changed

Reply:
Yes you are right, however, it should not change if we are just receiving block headers. Will keep it the same here but need to change it in `processTransaction`